pkg/tools/algorithm/roundrobin/roundrobin.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2020 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package roundrobin provides a selection by round robins algorithm
+package roundrobin
+
+import (
+	"sync/atomic"
+)
+
+// RoundRobin contains index
+type RoundRobin struct {
+	index uint32
```

Comment:
Can we use atomic and store size inside RoundRobin instead of passing it via parameter for function `Index`?
```suggestion
	index uint32
	size uint32
```

Reply:
If we do it like that, there will be a such a problem to use `RoundRobin` with mutable options list.